code/game/turfs/turf.dm

```
@@ -687,3 +687,22 @@
 		return A.has_gravity
 
 	return 1
+
+/turf/proc/set_area(area/A)
+	if(ispath(A))
+		var/path = A
+		A = locate(path)
+
+		if(!A)
+			A = new path
+	else if(!isarea(A))
+		return FALSE
```

IMO this should throw an exception instead.